diff --git a/build/tools/build-wxwidgets.py b/build/tools/build-wxwidgets.py
index 7777b40..b2a3cc3 100755
--- a/build/tools/build-wxwidgets.py
+++ b/build/tools/build-wxwidgets.py
@@ -271,7 +271,7 @@ def main(scriptName, args):
                             "--with-opengl",
                             "--enable-sound",
                             "--enable-graphics_ctx",
-                            "--enable-mediactrl",
+                            "--disable-mediactrl",
                             "--enable-display",
                             "--enable-geometry",
                             "--enable-debug_flag",
@@ -293,6 +293,7 @@ def main(scriptName, args):
         if sys.platform.startswith("darwin"):
             for xcodePath in getXcodePaths():
                 sdks = [
+                    xcodePath+"/SDKs/MacOSX10.9.sdk",
                     xcodePath+"/SDKs/MacOSX10.5.sdk",
                     xcodePath+"/SDKs/MacOSX10.6.sdk",
                     xcodePath+"/SDKs/MacOSX10.7.sdk",
@@ -652,4 +653,3 @@ def main(scriptName, args):
 if __name__ == '__main__':
     exitWithException = False  # use sys.exit instead
     main(sys.argv[0], sys.argv[1:])
-    
