diff --git a/configure b/configure
index 648ac49..a43d024 100755
--- a/configure
+++ b/configure
@@ -19466,7 +19466,7 @@ $as_echo "$as_me: WARNING: Could not determine deployment target from SDKSetting
         wxUSE_MACOSX_VERSION_MIN=
     fi
 elif test "x$wxUSE_MACOSX_VERSION_MIN" = "x"; then
-    wxUSE_MACOSX_VERSION_MIN=10.5
+    wxUSE_MACOSX_VERSION_MIN=10.12
 fi
 
 if test "x$MACOSX_SDK_OPTS" != "x"; then
@@ -42941,4 +42941,3 @@ echo "                                       libmspack          ${wxUSE_LIBMSPAC
 echo "                                       sdl                ${wxUSE_LIBSDL}"
 
 echo ""
-
